### ISSUE TITLE
Invalid search request

### ISSUE THREAD

**Issue by glombek** [azure-contrib/node-azure-search#18]
Updating to 0.0.12 from 0.0.11 causes the following error when initiating a search:

`{"error":{"code":"","message":"The request is invalid. Details: parameters : The parameter 'facet' in the request payload is not a valid parameter for the operation 'search'.\r\n"}}`

The generated request is a post to `https://[mysearchservice].search.windows.net/indexes/[index]/docs/search?api-version=2015-02-28-Preview` with no form data.

**Comment by dannav** [comment]
Search was changed from doing GET to performing a POST to the search endpoint

Property values may slightly differ for a post operation than from get. The following are the allowed properties for search doing a post.

in your case the parameter `facet` should be changed to `facets`.

```json
{  
     "count": true | false (default),  
     "facets": [ "facet_expression_1", "facet_expression_2", ... ],  
     "filter": "odata_filter_expression",  
     "highlight": "highlight_field_1, highlight_field_2, ...",  
     "highlightPreTag": "pre_tag",  
     "highlightPostTag": "post_tag",  
     "minimumCoverage": # (% of index that must be covered to declare query successful; default 100),  
     "orderby": "orderby_expression",  
     "scoringParameters": [ "scoring_parameter_1", "scoring_parameter_2", ... ],  
     "scoringProfile": "scoring_profile_name",  
     "search": "simple_query_expression",  
     "searchFields": "field_name_1, field_name_2, ...",  
     "searchMode": "any" (default) | "all",  
     "select": "field_name_1, field_name_2, ...",  
     "skip": # (default 0),  
     "top": #  
   }  
```

Hope that clarifies anything.

**Comment by richorama** [comment]
closed by #19
